### PR TITLE
fix(MessageBodyView): Reduce the number of redraw calls

### PR DESCRIPTION
**Pull request by adrien-coye**
Looking for issues with the rendering in `MessageBodyView` 
I was able to fix a bug and do some small improvements, reducing unnecessary view redraw.

### CODE REVIEW

File: Mail/Views/Thread/Message/InlineAttachmentWorker.swift [hunk 1]
```diff
@@ -34,7 +34,7 @@ final class InlineAttachmentWorker: ObservableObject {
     @Published var presentableBody: PresentableBody
 
     /// Set to true when done processing
-    @Published var isMessagePreprocessed: Bool
+    private var isMessagePreprocessed: Bool
```
Comment (adrien-coye) [inline Mail/Views/Thread/Message/InlineAttachmentWorker.swift:37]: We do not need to redraw once the `isMessagePreprocessed` is set to TRUE, the correct body is been processed.

File: Mail/Views/Thread/Message/MessageBodyView.swift [hunk 2]
```diff
@@ -45,23 +45,20 @@ struct MessageBodyView: View {
...
-                        loadBody(blockRemoteContent: blockRemoteContent)
-                    }
-                    .onChange(of: presentableBody) { _ in
-                        loadBody(blockRemoteContent: blockRemoteContent)
+                        loadBody(presentableBody: presentableBody, blockRemoteContent: blockRemoteContent)
                     }
-                    .onChange(of: isMessagePreprocessed) { _ in
-                        loadBody(blockRemoteContent: blockRemoteContent)
+                    .onChange(of: presentableBody) { newPresentableBody in
+                        loadBody(presentableBody: newPresentableBody, blockRemoteContent: blockRemoteContent)
```
Comment (adrien-coye) [inline Mail/Views/Thread/Message/MessageBodyView.swift:55]: Here using `newPresentableBody` fixed an issue where the processing was performed on the _previous_ value of presentableBody.

File: Mail/Views/Thread/WebView/WebViewModelOperations.swift [hunk 3]
```diff
@@ -18,10 +18,58 @@
...
 import CocoaLumberjackSwift
 import MailCore
+import RealmSwift
 import SwiftSoup
+import WebKit
 
-extension WebViewModel {
-    func loadHTMLString(value: String?, blockRemoteContent: Bool) async -> LoadResult {
+/// Something to process webview mutations on an isolated actor
+actor WebViewModelOperations {
```
Comment (adrien-coye) [inline Mail/Views/Thread/WebView/WebViewModelOperations.swift:26]: Locking webview mutations behind an actor.

**Review by adrien-coye — COMMENTED** [review]
Notes for reviewer.

File: Mail/Views/Thread/Message/MessageBodyView.swift [hunk 4]
```diff
@@ -32,7 +32,7 @@ struct MessageBodyView: View {
     @StateObject private var model = WebViewModel()
 
     let presentableBody: PresentableBody
-    let isMessagePreprocessed: Bool
+
     var blockRemoteContent: Bool
     @Binding var displayContentBlockedActionView: Bool
```
Comment (valentinperignon) [inline Mail/Views/Thread/Message/MessageBodyView.swift:37]: As you edit this view, could you re-order the properties in the right order? Binding, then let/var properties?

File: Mail/Views/Thread/Message/MessageBodyView.swift [hunk 5]
```diff
@@ -108,23 +105,26 @@ struct MessageBodyView: View {
         }
     }
 
-    private func loadBody(blockRemoteContent: Bool) {
+    private func loadBody(presentableBody: PresentableBody, blockRemoteContent: Bool) {
+        let uid = messageUid
```
Comment (valentinperignon) [inline Mail/Views/Thread/Message/MessageBodyView.swift:109]: Is there a reason to create the `uid` variable here?
If so, can we use a more explicit name to understand the reason why?

File: Mail/Views/Thread/Message/MessageBodyView.swift [hunk 6]
```diff
@@ -108,23 +105,26 @@ struct MessageBodyView: View {
...
             let loadResult = try await model.loadBody(
                 presentableBody: presentableBody,
                 blockRemoteContent: blockRemoteContent,
-                messageUid: messageUid
+                messageUid: uid
             )
 
-            displayContentBlockedActionView = (loadResult == .remoteContentBlocked)
+            Task { @MainActor in
+                displayContentBlockedActionView = (loadResult == .remoteContentBlocked)
```
Comment (valentinperignon) [inline Mail/Views/Thread/Message/MessageBodyView.swift:119]:
```suggestion
                displayContentBlockedActionView = loadResult == .remoteContentBlocked
```
We can use a more swifty syntax :)

File: Mail/Views/Thread/WebView/WebViewModel.swift [hunk 7]
```diff
@@ -37,6 +37,7 @@ final class WebViewModel: NSObject, ObservableObject {
     let contentBlocker: ContentBlocker
 
     let style: String = MessageWebViewUtils.generateCSS(for: .message)
```
Comment (valentinperignon) [inline Mail/Views/Thread/WebView/WebViewModel.swift:39]: This property is not necessary anymore as you declared it in the `WebViewModelOperations` actor

File: Mail/Views/Thread/WebView/WebViewModelOperations.swift [hunk 8]
```diff
@@ -18,10 +18,58 @@
...
+    init(webView: WKWebView, contentBlocker: ContentBlocker) {
+        self.webView = webView
+        self.contentBlocker = contentBlocker
+    }
+
+    func loadBody(presentableBody: PresentableBody,
+                  blockRemoteContent: Bool,
+                  messageUid: String,
+                  showBlockQuote: Bool) async throws -> WebViewModel
+        .LoadResult {
```
Comment (valentinperignon) [inline Mail/Views/Thread/WebView/WebViewModelOperations.swift:40]:
```suggestion
                  showBlockQuote: Bool) async throws -> WebViewModel.LoadResult {
```

**Review by valentinperignon — COMMENTED** [review]
Some of my messages never load :/
I tested by opening a long thread, and about 1 in 3 messages don't load.

**Comment by adrien-coye** [comment]
Since @valentinperignon shown this PR was not stable in some situations, I will not pursue it further.

**Pull request closed without merging**